LibWeb/SVG: Support painting cap_style and join_style for svg elements

mehrankamal opened the pull request:
This PR adds support for the `cap_style` and `join_style` in SVG painting

For testing out the changes, I had used below `html` sample

## Snippet:

```html
<html>
    <head></head>
    <body>
        <svg height="120" width="300" xmlns="http://www.w3.org/2000/svg">
            <g fill="none" stroke="red" stroke-width="16">
                <path stroke-linecap="butt" d="M10 20 l215 0"></path>
                <path stroke-linecap="round" d="M10 50 l215 0"></path>
                <path stroke-linecap="square" d="M10 80 l215 0"></path>
            </g>
            Sorry, your browser does not support inline SVG.
        </svg>
        <svg width="600" height="230" xmlns="http://www.w3.org/2000/svg">
            <polygon
                points="55,25 10,190 110,190"
                fill="lime"
                stroke="red"
                stroke-width="16"
                stroke-linejoin="round"
            ></polygon>
            <rect
                width="150"
                height="100"
                x="140"
                y="50"
                fill="yellow"
                stroke="red"
                stroke-width="16"
                stroke-linejoin="round"
            ></rect>
            Sorry, your browser does not support inline SVG.
        </svg>
        <svg width="600" height="230" xmlns="http://www.w3.org/2000/svg">
            <polygon
                points="55,25 10,190 110,190"
                fill="lime"
                stroke="red"
                stroke-width="16"
                stroke-linejoin="miter"
            ></polygon>
            <rect
                width="150"
                height="100"
                x="140"
                y="50"
                fill="yellow"
                stroke="red"
                stroke-width="16"
                stroke-linejoin="miter"
            ></rect>
            Sorry, your browser does not support inline SVG.
        </svg>
        <svg width="600" height="230" xmlns="http://www.w3.org/2000/svg">
            <polygon
                points="55,25 10,190 110,190"
                fill="lime"
                stroke="red"
                stroke-width="16"
                stroke-linejoin="bevel"
            ></polygon>
            <rect
                width="150"
                height="100"
                x="140"
                y="50"
                fill="yellow"
                stroke="red"
                stroke-width="16"
                stroke-linejoin="bevel"
            ></rect>
            Sorry, your browser does not support inline SVG.
        </svg>
    </body>
</html>
```

## Here's a Before/After regarding the changes

### Before:
<img width="1554" alt="Screenshot 2025-03-11 at 09 55 50" src="https://github.com/user-attachments/assets/84c21ea8-e200-4716-a019-2b388f93f23a" />

### After:
<img width="1554" alt="Screenshot 2025-03-11 at 09 56 20" src="https://github.com/user-attachments/assets/171e31dc-b940-4b48-a83f-b14284f92430" />

nico commented:
If you want, you can grab this as test: https://github.com/SerenityOS/serenity/blob/master/Tests/LibWeb/Screenshot/svg-stroke-styles.html

nico commented:
Per CI, you also have to update the expected screenshot for Screenshot/input/svg-background-no-natural-size.html as it's changing ever so slightly.

mehrankamal commented:
Hey @nico,

Thank you for the review. I have added the relevant test as you highlighted.

Also, I have updated the screenshot for for the `Screenshot/input/svg-background-no-natural-size.html` test using the `./Meta/ladybird.sh run headless-browser Tests/LibWeb/Screenshot/input/svg-background-no-natural-size.html -s 1` command, but upon visual inspection, it still seems to be the same image as the previous one.

Though, the rendering in the Ladybird application is different (with thinner border lines) and seems to be more correct. Let me know if I'm missing something?

AtkinsSJ commented:
The two tests you added are failing on CI. If you click through to the [Upload test artifacts](https://github.com/LadybirdBrowser/ladybird/actions/runs/13793590454/job/38579562865?pr=3900#step:19:1) step you can find a download link for the screenshots.

nico commented on Libraries/LibWeb/Painting/DisplayListPlayerSkia.cpp:
```diff
@@ -651,18 +651,48 @@ void DisplayListPlayerSkia::fill_path_using_paint_style(FillPathUsingPaintStyle
+        return SkPaint::kRound_Cap;
+    case Gfx::Path::CapStyle::Square:
+        return SkPaint::kSquare_Cap;
+    default:
```
not: I'd probably remove the default: branch and put VERIFY_NOT_REACHED() at the end instead. (Also below)
